File: backend/app/api/tasks.py
from fastapi import APIRouter
from app.schemas.tasks import TaskCreate, TaskUpdate
from app.core.security import CurrentUser, get_current_user
from fastapi import Depends
from app.services.task_service import create_task_service, update_task_service, delete_task_service, get_tasks_service

router = APIRouter(prefix="/api/tasks", tags=["tasks"])


@router.post("")
def create_task(
    task: TaskCreate,
    current_user: CurrentUser = Depends(get_current_user),
):
    return create_task_service(task, current_user["user_id"])

@router.put("/{task_id}")
def update_task(
    task_id: int,
    task: TaskUpdate,
    current_user: CurrentUser = Depends(get_current_user),
):
    return update_task_service(task_id, task, current_user["user_id"])

@router.delete("/{task_id}")
def delete_task(
    task_id: int,
    current_user: CurrentUser = Depends(get_current_user),
):
    return delete_task_service(task_id, current_user["user_id"])

# 查询
@router.get("")
def get_tasks(current_user: CurrentUser = Depends(get_current_user)):
    return get_tasks_service(current_user["user_id"])

File: frontend/src/api/tasks.ts
import request from '../utils/request'

export interface Task {
  id: number
  title: string
  description: string | null
  status: string
  created_at: string
  updated_at: string
}

export interface CreateTaskParams {
  title: string
  description?: string
}

export interface UpdateTaskParams {
  title?: string
  description?: string
  status?: string
}

export const getTasks = () => {
  return request.get<Task[]>('/tasks')
}

export const createTask = (data: CreateTaskParams) => {
  return request.post<Task>('/tasks', data)
}

export const updateTask = (
  id: number,
  data: UpdateTaskParams,
) => {
  return request.put<Task>(`/tasks/${id}`, data)
}

export const deleteTask = (id: number) => {
  return request.delete(`/tasks/${id}`)
}